src/unix/netbsd.c

```diff
@@ -79,19 +73,14 @@ int uv_exepath(char* buffer, size_t* size) {
   mib[1] = KERN_PROC_ARGS;
   mib[2] = -1;
   mib[3] = KERN_PROC_PATHNAME;
-  int_size = ARRAY_SIZE(int_buf);
 
-  if (sysctl(mib, 4, int_buf, &int_size, NULL, 0))
+  if (sysctl(mib, ARRAY_SIZE(mib), buffer, size, NULL, 0) != 0 && errno != ENOSPC)
     return UV__ERR(errno);
 
-  /* Copy string from the intermediate buffer to outer one with appropriate
-   * length.
-   */
-  /* TODO(bnoordhuis) Check uv__strscpy() return value. */
-  uv__strscpy(buffer, int_buf, *size);
-
-  /* Set new size. */
-  *size = strlen(buffer);
+  /* In case the buffer might not be C-string complete, adding trailing end */
+  if (!errno)
```

Comment:
Does `sysctl()` _always_ set errno, even in the happy path?

Reply:
No. Here it will set errno only for a truncated transfer (ENOSPC).

Reply:
Actually there could a corner case that a floating errno ENOSPC could cause this to be triggered.

There are two options:

 - set errno to 0 before sysctl()
 - check for `ENOSPC` in this condition

Probably the former is bit cleaner as at least it will reset leaked errno from an unspecified place.

Reply:
I prefer to set errno before hand personally, will take that route.